File: app/src/main/java/fr/free/nrw/commons/wikidata/mwapi/UserInfo.java
package fr.free.nrw.commons.wikidata.mwapi;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import java.util.Map;


public class UserInfo {
    @NonNull private String name;
    @NonNull private int id;

    //Block information
    private int blockid;
    private String blockedby;
    private int blockedbyid;
    private String blockreason;
    private String blocktimestamp;
    private String blockexpiry;

    // Object type is any JSON type.
    @SuppressWarnings("MismatchedQueryAndUpdateOfCollection")
    @Nullable private Map<String, ?> options;

    public int id() {
        return id;
    }

    @NonNull
    public String blockexpiry() {
        if (blockexpiry != null)
            return blockexpiry;
        else return "";
    }
}

File: app/src/main/java/fr/free/nrw/commons/wikidata/mwapi/UserInfo.kt
package fr.free.nrw.commons.wikidata.mwapi

data class UserInfo(
    val name: String = "",
    val id: Int = 0,

    //Block information
    val blockid: Int = 0,
    val blockedby: String? = null,
    val blockedbyid: Int = 0,
    val blockreason: String? = null,
    val blocktimestamp: String? = null,
    val blockexpiry: String? = null,

    // Object type is any JSON type.
    val options: Map<String, *>? = null
) {
    fun id(): Int = id

    fun blockexpiry(): String = blockexpiry ?: ""
}
